crates/uv-interpreter/src/virtualenv.rs

```diff
@@ -88,6 +89,12 @@ pub(crate) fn virtualenv_from_working_dir() -> Result<Option<PathBuf>, Error> {
             debug!("Found a virtualenv named .venv at: {}", dot_venv.display());
             return Ok(Some(dot_venv));
         }
+
+        // Otherwise, if we're _within_ a virtualenv, return it.
+        if dir.join("pyvenv.cfg").is_file() {
+            debug!("Found a virtualenv at: {}", dir.display());
+            return Ok(Some(dir.to_path_buf()));
+        }
     }
```

Comment:
The main thing I wanted to change here was looking in ancestor directories, so that if you're in `.venv/bin`, we still find the current venv. Whether this check comes before or after the `// Search for a `.venv` directory.` step is somewhat less clear, that's a weird situation (you have a nested venv within a venv).

Reply:
https://github.com/astral-sh/uv/pull/3406